File: clojure/log_levels.clj
(ns log-levels
  (:require [clojure.string :as str]))

(defn message
  [s]
  (str/trim (str/replace-first s #"\[(ERROR|INFO|WARNING)\]:" "")))

(defn log-level
  [s]
  (cond (str/starts-with? s "[ERROR]") "error"
        (str/starts-with? s "[INFO]") "info"
        (str/starts-with? s "[WARNING]") "warning"))

(defn reformat [s] (format "%s (%s)" (message s) (log-level s)))

File: cs/LogLevels.cs
using System;

static class LogLine
{
    public static string Message(string logLine)
    {
        int i = logLine.IndexOf(" ");
        return logLine.Substring(i + 1).Trim();
    }

    public static string LogLevel(string logLine)
    {
        int i = logLine.IndexOf("]");
        return logLine.Substring(1, i - 1).ToLower();
    }

    public static string Reformat(string logLine)
    {
        return $"{Message(logLine)} ({LogLevel(logLine)})";
    }
}
